From 7f63576ecd62e027c3f43a7a8a363eb604c3363c Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Thu, 3 Aug 2023 13:19:47 +0200
Subject: [PATCH 463/464] arm64: dts: rk3588-orangepi-5-plus: Add fan control
 node

The board has a fan connector. There are still some issues:

https://github.com/megous/linux/pull/22

Signed-off-by: Tom Briden <tom@decompile.me.uk>
Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 .../boot/dts/rockchip/rk3588-orangepi-5-plus.dts    | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
index 86c7150e3898..668e9192bcad 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dts
@@ -66,8 +66,14 @@ headphone_amp: audio-amplifier1 {
 		sound-name-prefix = "Headphones Amp";
 	};
 
-	/*XXX: fan */
 	/* GPIO3 B2, active high, PWM3_IR_M1 */
+	fan: pwm-fan {
+		compatible = "pwm-fan";
+		cooling-levels = <0 95 145 195 255>;
+		fan-supply = <&vcc5v0_sys>;
+		pwms = <&pwm3 0 50000 0>;
+		#cooling-cells = <2>;
+	};
 
 	ir-receiver {
 		compatible = "gpio-ir-receiver";
@@ -381,6 +387,11 @@ ir_receiver_pin: ir-receiver-pin {
 	};
 };
 
+&pwm3 {
+	status = "okay";
+	pinctrl-0 = <&pwm3m1_pins>;
+};
+
 &saradc {
 	status = "okay";
 	vref-supply = <&vcc_1v8_s0>;
-- 
2.34.1

